delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeViewDef.scala

```diff
@@ -554,4 +564,11 @@ object CompositeViewDef {
     } yield target.id -> result
   }
 
+  /** Union of all resourceTypes specified in the view's projections */
+  private def projectionTypes(view: ActiveViewDef): Set[Iri] = {
+    val targets = view.value.projections
+    if (targets.exists(_.resourceTypes.isEmpty)) Set.empty[Iri]
+    else targets.foldLeft(Set.empty[Iri])(_ ++ _.resourceTypes)
+  }
```

Comment:
The model here where empty set means there is no restriction is quite confusing. Consider changing it to be an ADT such as `NoRestriction` and `RestrictedTo(types: Set[Iri])`

Reply:
=> #4374